ee/server/service/teams.go

```diff
@@ -1064,8 +1066,12 @@ func (svc *Service) editTeamFromSpec(
 	}
 	team.Config.MDM.MacOSSetup.EnableEndUserAuthentication = spec.MDM.MacOSSetup.EnableEndUserAuthentication
 
+	windowsEnabledAndConfigured := appCfg.MDM.WindowsEnabledAndConfigured
+	if opts.DryRunAssumptions != nil && opts.DryRunAssumptions.WindowsEnabledAndConfigured.Valid {
+		windowsEnabledAndConfigured = opts.DryRunAssumptions.WindowsEnabledAndConfigured.Value
+	}
```

Comment:
Sometimes, running dry run on team must assume that some setting(s) will be set by the global config. But they haven't actually been set yet, since we are doing a dry run.
I've created a new pattern to pass in such assumptions in the `DryRunAssumptions` struct.